src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
+        this.state.searchTerm === ""
+          ? `https://icanhazdadjoke.com/search`
+          : `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`;
+      const fetchUrl = fetch(url, {
+        method: "GET",
+        headers: { accept: "application/json" }
+      });
+      const response = await fetchUrl;
+      const myData = await response.json();
+      this.setState({
+        jokeData: { results: myData.results },
+        loadingState: "loaded"
+      });
+    } catch (e) {
+      this.setState({ loadingState: "Something went wrong" });
```

Comment:
You're overloading `loadingState` here. Everywhere else, it's the status of the fetch call, but here it's a message you're likely displaying to the user. I think you should make the loading state `"error"` or something and then display a message in `render` based on `this.state.loadingState`.